pkg/internal/deprecated/override.go

```diff
@@ -0,0 +1,89 @@
+/*
+Copyright 2021 The Tekton Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package deprecated
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/tektoncd/pipeline/pkg/apis/config"
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline"
+	"github.com/tektoncd/pipeline/pkg/pod"
+	corev1 "k8s.io/api/core/v1"
+)
+
+func NewOverrideWorkingDirTransformer(ctx context.Context) pod.Transformer {
+	return func(p *corev1.Pod) (*corev1.Pod, error) {
+		if shouldOverrideWorkingDir(ctx) {
+			fmt.Println("Let's overwrite !!!!")
+			for i, c := range p.Spec.Containers {
+				// FIXME(vdemeester) use a function for `step-` prefix
```

Comment:
There's a func we already publicly export for this in ... \<shuffles papers> ... `pod.IsContainerStep()` from `pkg/pod/entrypoint.go`.